### ISSUE TITLE
Improve Error messaging for GeoODK submission responses

### ISSUE THREAD

**Issue by linzjax** [Cadasta/cadasta-platform#432]
Currently the user is only getting a `<file-name> -- Success` or a garbled system error. The messaging should be improved so that the user actually knows what went wrong and how to fix it.

**Comment by ian-ross** [comment]
@linzjax What's the status of this one now that your XForms submissions stuff has been merged?  I know you wanted to make the error messaging better, but do you think we can close this or move it to the backlog, since it's really only a problem because of the interim "users edit XLSForms" approach to questionnaires?

**Comment by linzjax** [comment]
I would move it to the backlog (or close it for now since it's not technically a problem with our platform). I think since we're going through a handholding phase with those building xforms themselves, we'll be available to help work out kinks. Maybe one day we can actually update the messaging in geoodk

**Comment by bjohare** [comment]
Need to add logging of submission errors.

**Comment by dpalomino** [comment]
This should be fixed with @jnordling fix. Needs some testing.

**Comment by linzjax** [comment]
@dpalomino: FYI: Once the update is integrated into ODK, we'll need to revisit the error messages being sent. Most of them are not clear or optimized for real users.

**Comment by wonderchook** [comment]
@linzjax this is already in GeoODK, I believe the next ODK release is in a few weeks as well. 

@dpalomino this should be a task for a sprint.

**Comment by dpalomino** [comment]
@linzjax @wonderchook 
ok, added in the backlog for being tracked. Are those messages available in a single file or similar, so we can review them? If they are, I can take a look and make some suggestions.

**Comment by linzjax** [comment]
Not really. Currently, they're whatever error message django throws with `Location error:`, `Party error:` etc. in front of them. It might be worth dividing up the code so that errors are thrown at more specific times so that we can give clearer feedback. Right now, if _anything_ in the create_<record> process breaks, it throws one error, so it's hard to figure out where in the process things went wrong.

**Comment by wonderchook** [comment]
@SteadyCadence I'm thinking we could prioritize improving this. Any specific issues that you think exact errors could help?

**Comment by SteadyCadence** [comment]
The AttributeType error is a big one that could be more specific. I find that occurs when required fields are not in the proper order (like party_type in the party group or a resource nor named properly). I can get more specific later today.
